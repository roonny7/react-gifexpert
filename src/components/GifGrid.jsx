import React, { useEffect, useState } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem} from './GifItem';





export const GifGrid = ( { category }) => {

    const { images, isLoading } = useFetchGifs(category);

  console.log(images, isLoading)
    
    

    return (
      <>
        <h3>{ category } </h3>
        {
            isLoading
            ? (<h2>Cargando</h2>)
            : null

        }
        <h2>{ isLoading}</h2>
        <div className="card-grid">
            {
              images.map( (img) => (
                //<GifItem key={img.id}  title={img.title} url={img.url}/>
                <GifItem key={img.id} {...img}/>

                
              ))


            }            
        </div>
      </>
    )
}
