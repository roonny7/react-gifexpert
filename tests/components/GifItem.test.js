import { render, screen } from "@testing-library/react"
import { exact } from "prop-types";
import { GifItem } from "../../src/components/GifItem"

describe('pruebas en GifItem', () => { 

    const title ="Metal gear";
    const url = 'https://metalgear.com/';
    test('Debe hacer match con el snapshot', () => { 

        const {container} = render(<GifItem title={title} url={url}/>)  
        expect(container).toMatchSnapshot();

    })

    test('Debe de mostrar la imagen con el url y el alt indicado', () => { 
        render(<GifItem title={title} url={url} />);
        //screen.debug();
        //expect(screen.getByRole('img').src).toEqual(url);
        //expect(screen.getByRole('img').alt).toEqual(title);

        const { src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
         
    })

    test('Debe de mostrar el tÃ­tulo en el componente', () => { 
        render(<GifItem title={title} url={url} />)
        expect(screen.getByText(title)).toBeTruthy();
     })


 })