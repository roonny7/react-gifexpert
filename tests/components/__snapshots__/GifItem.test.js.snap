// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Metal gear"
      src="https://metalgear.com/"
    />
    <p>
      Metal gear
    </p>
  </div>
</div>
`;
